packages/oats-runtime/src/runtime.ts

```diff
@@ -18,6 +22,23 @@ const tagKey = Symbol();
 
 type ScalarWithoutBrand<V> = V extends ShapedClass<infer S> ? ScalarWithoutBrand<S> : V;
 
+// helper to set necessary metadata  for valueclass instances
+export function instanceAssign(
+  to: ValueClass,
+  value: Record<string, unknown>,
+  opts: make.MakeOptions | { unSafeSet: boolean } | undefined,
+  builder: make.Maker<any, any>
+) {
+  Object.assign(to, opts && 'unSafeSet' in opts ? value : builder(value, opts).success());
+  // when we create an instance of a ValueClass we need to propagate the type information from 'value' to the newly
+  // constructed instance
+  const existingType = typeTag.getType(value);
```

Comment:
the input value may have a reflection type already so need to merge that with the current instances type.